### DISCUSSION TITLE
Jamulus 3.8.0 and opus-git

### DISCUSSION THREAD

**Discussion by deesnook** [jamulussoftware/jamulus#1820] [Technical Support & Help] [answered]
Hi there,

I was just about to update to v3.8.0-1 on Manjaro Linux when I noticed that Jamulus now uses opus-git 0.0-1 from AUR rather than opus 1.3.1-2 from the official repositories as it did before. Since other programs (notably Chromium) are depending on the latter, which would be replaced when updating Jamulus, I was wondering if this could cause any issues and what is the reason for switching to opus-git

Thanks!

**Comment by jujudusud** [comment]
As one of the Archlinux Jamulus package maintainer, I will be happy to discuss with you any problems you will have using it on your system.
I will have a look at the differences between the two versions of the OPUS package but I think we have to use it as soon as Jamulus requiers a custom build (provided by the git version) that will never be merged in OPUS official release.
Regards.

**Reply by deesnook** [reply]
Thanks, I'll try the update then and will let you know if I run into any problems

**Reply by hoffie** [reply]
> I think we have to use it as soon as Jamulus requiers a custom build (provided by the one) that will never be merged in OPUS official release.

IIRC, Jamulus basically uses the official libopus. However, we bundle the latest release (1.3.1 from 2019) which contains an issue which affects applications with a specific opus configuration. Jamulus happens to use this configuration. See  for details. Therefore, Jamulus' bundled libopus contains fixes for that:
https://github.com/xiph/opus/issues/172
https://gitlab.xiph.org/xiph/opus/-/issues/2323 (#323)

In other words: If there was a new opus release (1.3.2 or something) with these officially merged patches, Jamulus should work with vanilla opus as well. Distributions could probably backport those patches into their official 1.3.1 releases with a distribution-specific revision bump.

**Reply by ghost** [reply]
The OPUS package in the official Jamulus is only used by the official Jamulus that contains the OPUS package. Furthermore, the official Jamulus does not use any other OPUS.
If this is not the case, something is very wrong.

**Reply by jujudusud** [reply]
@deesnook , Remembering a discussion with opus IRC users, I think I have to check if it is possible to use Jamulus with `opus` instead of `opus-git` on Archlinux. -> After a test, Jamulus don't work with `opus` package, only with `opus-git`.

@DavidSavinkoff , there is a `qmake "CONFIG+=opus_shared_lib" Jamulus.pro` option to use system wide OPUS library.

@hoffie , some info of my discussion on #opus irc channel:

`
jujudusud: Hi all, First time I come here, I'm jujudusud, french maintener of Jamulus AUR package for Archlinux. I want to use Jamulus with system wide opus librairy. Opus is at 1.3.1 version on Archlinux repository. With Jamulus there is an decoder assertion which make Jamulus crash. The "problem" is solved installing git version (which includes : https://gitlab.xiph.org/xiph/opus/-/commit/06a651547f4a197c29b6abc23b530833855ee358 ).
Do you plan to release a new version of opus ?

TD-Linux: yes, however OPUS_CUSTOM isn't normally part of a system-wide libopus library, so you still will need a local build for your package

jujudusud: Hi TD-Linux, thanks for your rapid answer. To better understand what you said (my french poor english :-) ) Jamulus would use a custom version of opus. Ok, I will have a look at that, questionning Jamulus dev if a costum is necessary, why and how.

TD-Linux: jujudusud, actually I checked and arch's PKGBUILD does enable custom modes. I'm not sure if that's a good idea in general but it does mean you can use the system one when 1.3.2 is out

jujudusud: For the AUR git version ? -> Yes, ok it's the AUR opus-git package. Ok it is the same extra/x86_64/opus package. understood all the things you said TD-Linux . thanks a lot for your help and the informations. :-)

jujudusud: Is a release 1.3.2 of opus planned ?
`

No more answers ...

**Reply by hoffie** [reply]
>  Is a release 1.3.2 of opus planned?

That's the relevant question. :)
Given that there have been bug fixes which are relevant (at least to Jamulus), it would be good to have such a release. On the other hand -- as said -- you might be able to convince Arch's opus maintainer to provide a 1.3.1-1234 build which backports the relevant patches. At least in my times as packager for Gentoo, this was a common thing to do -- backport upstream patches if they are relevant to the distribution, even if there are no upstream releases with the fixes yet.

I still like the typical distribution approach of avoiding bundled libraries (e.g. avoid building Jamulus with the packaged opus, but use the distribution version instead). This has certain benefits, especially regarding maintenance/security updates. On the other hand, it has downsides (as can be seen here as well) and there are tendencies to do stuff different (thinking of Go which seems to favor vendor'ing libs and thinking of containers which bundle and duplicate all kind of "stuff" in a fire and forget approach... ;).

**Reply by jujudusud** [reply]
@deesnook 
Any news here ? Did you install opus-git and is it ok with all other applications on your system ?

**Reply by deesnook** [reply]
Yes, I installed it and so far it seems to work fine. At least audio in Chromium has worked as before. However, since I don't know what opus is used for exactly, I couldn't tell if I have checked all use cases. I'll definitely let you know if something weird comes up

**Reply by jujudusud** [reply]
opus is an interactive audio codec. Basicly explained it is the part of the software which encode your sound to reduce its size before going through the internet and decode the sound from the server on your machine to playback on the speakers.
[Opus website](https://opus-codec.org/)

**Comment by jujudusud** [comment, marked as answer]
AUR package no longer ue opus-git package.
Should close this discussion.
